### ISSUE TITLE
Connection refused to swarm registry

### ISSUE THREAD

**Issue by chumpyy** [swarmlet/swarmlet#27]
Awesome project, am really looking forward to the future improvements!

Currently, when pushing a deployment with a Dockerfile, I systematically get a 
"Get http://127.0.0.1:5000/v2/:  dial tcp 127.0.0.1:5000: connect: connection refused"

Same message if I do a "docker-compose push" (following a successful build) on the server itself.

Haven't touched consul's configuration - what am I missing?

Many thanks and keep up the great work!

**Comment by chumpyy** [comment]
So yeah, my bad - I assumed the registry was deployed by default along with the loadbalancer and co, but in my case it wasn't.
Got it deployed as suggested internally, without exposure and without security.
I'll take a look at securing it and configuring swarmpit to link to it in the registries section.

Sorry for the unnecessary noise

**Comment by woudsma** [comment]
Thanks for your interest in the project!

What exactly happened that the registry wasn't deployed as well during installation?
I'm curious, that should not happen 😄

And yeah, contributions are most welcome! I'm always running out of time.
I've just refactored the master branch a bit, that should improve some logic and readability.
Next on the list is the registry, I feel a bit frustrated that I haven't figured that out yet.
Any kind of feedback is much appreciated.

I was looking at providing a simple frontend for the registry as well, a simple UI that works with the `registry:2` image. This looked interesting and is probably easy to set up: https://github.com/kwk/docker-registry-frontend

Today I added Portainer as an optional service, that was ridiculously easy. I spent 10 minutes writing a `docker-compose.yml` file and Portainer was up and running at https://portainer.swarmlet.dev (with automatic SSL) on the first git push 🥳 

If you have any suggestions on how to implement the exposed + authenticated registry, let me know! I still have to do some research before I'll start.
